kubernetes/deployments_kubeclient.go

```diff
@@ -480,6 +482,29 @@ func (kc *kubeClient) GetDeploymentStatSeries(spaceName string, appName string,
 	return result, nil
 }
 
+func (kc *kubeClient) DeleteDeployment(spaceName string, appName string, envName string) error {
+	envNS, err := kc.getEnvironmentNamespace(envName)
+	if err != nil {
+		return errs.WithStack(err)
+	}
+	// Delete routes
+	err = kc.deleteRoutes(appName, envNS)
+	if err != nil {
+		return err
+	}
+	// Delete services
+	err = kc.deleteServices(appName, envNS)
```

Comment:
If this call fails for whatever reason, calling DeleteDeployment again, will it then fail in deleteRoutes due to 404 or similar?

(it seems like it just wouldn't find any routes to delete, but no test case to prove it.. ;)

Reply:
If there are no routes matching the expected "app" label, none will be deleted and no error is returned. The same is true for services. If the deployment config is missing, an error will be returned. Since the deployment config is the last to be deleted, failure to delete routes or services will still allow everything to be deleted as expected with a subsequent call(s) to DeleteDeployment.

I have added test cases for missing routes, services and deployment configs in my tests branch: https://github.com/ebaron/fabric8-wit/commit/75049a9576ac059c6ded000cff17dddc777b35fc
Thanks for the suggestion!